File: app/src/main/java/com/popalay/cardme/presentation/base/ViewModelView.java
package com.popalay.cardme.presentation.base;

import com.arellomobile.mvp.viewstate.strategy.SingleStateStrategy;
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;

public interface ViewModelView<M extends BaseViewModel> extends BaseView {

    @StateStrategyType(SingleStateStrategy.class)
    void setViewModel(M viewModel);
}

File: app/src/main/java/com/popalay/cardme/presentation/base/ViewModelView.kt
package com.popalay.cardme.presentation.base

import com.arellomobile.mvp.viewstate.strategy.SingleStateStrategy
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType

interface ViewModelView<in M : BaseViewModel> : BaseView {

    @StateStrategyType(SingleStateStrategy::class)
    fun setViewModel(viewModel: M)
}
